pkg/scheduler/context.go

```diff
@@ -626,27 +656,18 @@ func (cc *ClusterContext) processAsks(request *si.UpdateRequest) {
 			continue
 		}
 
-		// if app info doesn't exist, reject the request
-		app := partition.getApplication(siAsk.ApplicationID)
-		if app == nil {
-			msg := fmt.Sprintf("Failed to find application %s, for allocation %s", siAsk.ApplicationID, siAsk.AllocationKey)
-			log.Logger().Info(msg)
-			rejectedAsks = append(rejectedAsks,
```

Comment:
This has all been folded into one call. Instead of first checking the app on the partition and then calling the app directly we call the add on the partition which calls the add on the app. This brings it inline with the other checks we do and the siAsk is the loop variable and has not changed. The error still exists and still gets [returned](https://github.com/apache/incubator-yunikorn-core/pull/251/files)